egi_notebooks_hub/egiauthenticator.py

```diff
@@ -193,28 +174,30 @@ class DataHubAuthenticator(EGICheckinAuthenticator):
 
     map_users = Bool(False, config=True, help="""perform mapping""")
 
-    token_name = Unicode(
-        "notebooks.egi.eu",
+    oneclient_token_name = Unicode(
+        "oneclient.notebooks.egi.eu",
+        config=True,
+        help="""Name of token in the onezone for oneclient""",
+    )
+
+    onezone_token_name = Unicode(
+        "onezone.notebooks.egi.eu",
         config=True,
-        help="""Name of token in the onezone for the user""",
+        help="""Name of token in the onezone for the onezone""",
     )
 
     storage_id = Unicode("", config=True, help="""Storage id to use for mapping""")
 
-    async def authenticate(self, handler, data=None):
-        user_data = await super(DataHubAuthenticator, self).authenticate(handler, data)
-        http_client = AsyncHTTPClient()
+    async def create_onedata_token(self, checkin_token, token_name, caveats):
         onedata_token = None
         onedata_user = None
-        # We now go to the datahub to get a token
-        checkin_token = user_data["auth_state"]["access_token"]
+        http_client = AsyncHTTPClient()
         headers = {
             "content-type": "application/json",
             "x-auth-token": "egi:%s" % checkin_token,
```

Comment:
should this be configurable as welll? imagine deployments of Onedata not configured with checkin ( e.g.  Panosc)

Reply:
ok, rethinking this

Reply:
take a look now. I also moved things to a separate onedata.py file